pubsub/jetstream/metadata.go

```diff
@@ -119,5 +123,15 @@ func parseMetadata(psm pubsub.Metadata) (metadata, error) {
 		m.internalAckPolicy = nats.AckExplicitPolicy
 	}
 
+	// Explicit check to prevent overriding the Single default
+	// (the previous behavior) if not set.
+	if psm.Properties[pubsub.ConcurrencyKey] != "" {
```

Comment:
This needs to be removed too? Still not case-insensitive

Reply:
The `pubsub.Concurrency` function defaults to `pubsub.Parallel` when the value is empty/undefined. So if I remove this check, the function will default to `Parallel` which I want to prevent in this case since the previous behavior was not that.

Reply:
And FWIW, it appears the [`Concurrency`](https://github.com/dapr/components-contrib/blob/main/pubsub/concurrency.go) does appear to check for case-insensitivity.

Reply:
Ok, that's a bug we need to fix in the Concurrency function. Both the fact that it overrides the default value in the object if the metadata property is empty (it should respect what you set in `metadata{}`), and the fact that it isn't case-insensitive.

For now, I'll accept this, but can you add a comment with a TODO linking to this conversation please?

Reply:
Added the TODO comment, so ready for merge from my perspective.